apps/ui/src/helpers/utils.ts

```diff
@@ -404,9 +404,26 @@ export function simplifyURL(fullURL: string): string {
   return `${url.hostname}${url.pathname.replace(/\/$/, '')}`;
 }
 
-export function getChoiceText(availableChoices: string[], choice: number | number[]) {
+export function getChoiceWeight(selectedChoices: Record<string, number>, index: number) {
+  const whole = Object.values(selectedChoices).reduce((a, b) => a + b, 0);
+  const percent = selectedChoices[index + 1] / whole;
+
+  return isNaN(percent) ? 0 : percent;
+}
+
+export function getChoiceText(
+  availableChoices: string[],
+  choice: number | number[] | Record<string, number>
+) {
   if (Array.isArray(choice)) {
     return choice.map(index => availableChoices[index - 1]).join(', ');
+  } else if (typeof choice === 'object') {
```

Comment:
```suggestion
  }

   if (typeof choice === 'object') {
```

I've been working on it on separate issue but then noticed this will be handled by this PR and came up with this function:
```js
export function getChoiceText(
  availableChoices: string[],
  choice: number | number[] | Record<number, number>
) {
  if (typeof choice === 'number') {
    return availableChoices[choice - 1];
  }

  if (Array.isArray(choice)) {
    return choice.map(index => availableChoices[index - 1]).join(', ');
  }

  const total = Object.values(choice).reduce((acc, percentage) => acc + percentage, 0);

  return Object.entries(choice)
    .filter(([, percentage]) => percentage > 0)
    .map(
      ([index, percentage]) =>
        `${_p(percentage / total)} for ${availableChoices[Number(index) - 1]}`
    )
    .join(', ');
}
```

Benefits with this is that it avoids computing total for every choice and also filters out options with 0% weight (same as v1).

Reply:
Implemented your idea, as well as adding missing `Record<string, number>` types for new weighted choices